sdk/storage/storage-blob/review/generated/src/models/index.d.ts

```diff
@@ -0,0 +1,6713 @@
+    incrementalCopy?: boolean;
+    destinationSnapshot?: string;
+    deletedTime?: Date;
+    remainingRetentionDays?: number;
+    /**
+     * Possible values include: 'P4', 'P6', 'P10', 'P15', 'P20', 'P30', 'P40', 'P50', 'P60', 'P70',
+     * 'P80', 'Hot', 'Cool', 'Archive'
+     */
+    accessTier?: AccessTier;
+    accessTierInferred?: boolean;
+    /**
+     * Possible values include: 'rehydrate-pending-to-hot', 'rehydrate-pending-to-cool'
+     */
+    archiveStatus?: ArchiveStatus;
+    customerProvidedKeySha256?: string;
+    accessTierChangeTime?: Date;
+}
+/**
+ * An interface representing BlobMetadata.
+ */
+export interface BlobMetadata {
+    encrypted?: string;
+    /**
+     * Describes unknown properties. The value of an unknown property MUST be of type "string". Due
+     * to valid TS constraints we have modeled this as a union of `string | any`.
+     */
+    [property: string]: string | any;
+}
+/**
+ * An Azure Storage blob
+ */
+export interface BlobItem {
+    name: string;
+    deleted: boolean;
+    snapshot: string;
+    properties: BlobProperties;
+    metadata?: BlobMetadata;
+}
+/**
+ * An interface representing BlobFlatListSegment.
+ */
+export interface BlobFlatListSegment {
+    blobItems: BlobItem[];
+}
+/**
+ * An enumeration of blobs
+ */
+export interface ListBlobsFlatSegmentResponse {
+    serviceEndpoint: string;
+    containerName: string;
+    prefix?: string;
+    marker?: string;
+    maxResults?: number;
+    delimiter?: string;
+    segment: BlobFlatListSegment;
+    nextMarker?: string;
+}
+/**
+ * An interface representing BlobPrefix.
+ */
+export interface BlobPrefix {
+    name: string;
+}
+/**
+ * An interface representing BlobHierarchyListSegment.
+ */
+export interface BlobHierarchyListSegment {
+    blobPrefixes?: BlobPrefix[];
+    blobItems: BlobItem[];
+}
+/**
+ * An enumeration of blobs
+ */
+export interface ListBlobsHierarchySegmentResponse {
+    serviceEndpoint: string;
+    containerName: string;
+    prefix?: string;
+    marker?: string;
+    maxResults?: number;
+    delimiter?: string;
+    segment: BlobHierarchyListSegment;
+    nextMarker?: string;
+}
+/**
+ * Represents a single block in a block blob.  It describes the block's ID and size.
+ */
+export interface Block {
+    /**
+     * The base64 encoded block ID.
+     */
+    name: string;
+    /**
+     * The block size in bytes.
+     */
+    size: number;
+}
+/**
+ * An interface representing BlockList.
+ */
+export interface BlockList {
+    committedBlocks?: Block[];
+    uncommittedBlocks?: Block[];
+}
+/**
+ * An interface representing BlockLookupList.
+ */
+export interface BlockLookupList {
+    committed?: string[];
+    uncommitted?: string[];
+    latest?: string[];
+}
+/**
+ * Properties of a container
+ */
+export interface ContainerProperties {
+    lastModified: Date;
+    etag: string;
+    /**
+     * Possible values include: 'locked', 'unlocked'
+     */
+    leaseStatus?: LeaseStatusType;
```

Comment:
I'm not a TS expert but I think many things are '?' which do not have to be, like leaseStatus

Reply:
No quite understand your requestion. If you refer to the question mark following "leaseStatus". It means this property can be undefined.

Reply:
We also pass this to `setProperties()` so `?` allows us to only specify some of these fields.

Reply:
You could have setProperties take a Partial<> of the interface if you wanted to allow that